xkcd/pkg/web/web.go

```diff
@@ -0,0 +1,60 @@
+package web
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/vespian/go-exercises/xkcd/pkg/types"
+)
+
+func fetchStory(url string) (types.Story, error) {
+	var result types.Story
+
+	fmt.Printf("Fetching url %s\n", url)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return result, err
+	}
+
+	defer func() {
+		resp.Body.Close()
+	}()
+
+	if resp.StatusCode == http.StatusNotFound {
+		return types.Story{}, nil
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return result, fmt.Errorf("fetching story %s failed: %s", url, resp.Status)
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
+		return result, fmt.Errorf("decoding story %s failed: %s", url, err)
+	}
+
+	return result, nil
+}
+
+func Fetch(format string, rg types.Range) (types.AllStories, error) {
+	res := make(types.AllStories)
+
+	for i := rg.Min; i < rg.Max; i++ {
+		url := fmt.Sprintf(format, i)
+		story, err := fetchStory(url)
+		if err == nil {
+			if story != (types.Story{}) {
```

Comment:
if you used pointers, this would look a lot better, you could then do `if story != nil` which reads much better than this.